feature: soft delete workspace

danhemerlein opened the pull request:
This adds UI and an API endpoint Delete operation.
The Delete operation soft deletes workspaces only if they have no connections. 

This also adds the `confirmOpWithInput` which requires triggers a confirmation modal requiring the user input a specific string before enabling the 'OK' button.

Please see this screen recording demonstrating the new functionality:
[https://share.zight.com/qGu8lrPD](https://share.zight.com/qGu8lrPD)

A big thing to keep in mind here is that database records are not actually deleted - they're soft deleted in the database
    
<!-- This is an auto-generated description by mrge. -->
---

## Summary by mrge
Added the ability to delete a workspace from the settings page, with a confirmation flow and backend support. Workspaces can only be deleted if they have no active connections.

- **New Features**
  - "Delete workspace" button in the workspace settings, with a multi-step confirmation and required input.
  - API endpoint to soft delete a workspace if it has no connections.

<!-- End of auto-generated description by mrge. -->

danhemerlein committed:
add UI for workspace deletion

danhemerlein committed:
update Jitsu Button to accept className as a prop

danhemerlein committed:
center button

danhemerlein committed:
implement workspace soft deletion

danhemerlein committed:
update copy, updatedAt and redirection logic

danhemerlein committed:
handle only one active workspace

danhemerlein committed:
address code review from merge io

danhemerlein commented on webapps/console/components/JitsuButton/JitsuButton.tsx:
```diff
@@ -10,6 +10,7 @@ import { useWorkspace } from "../../lib/context";
 export type JitsuButtonProps = ButtonProps & {
   //set to true if href is relative workspace link
   ws?: boolean;
+  className?: string;
```
this allows for additional css classes to be added to buttons

vklimontovich commented:
Overall, it looks good, thank you very much for the contribution!

But we can make it even better by addressing a few issues:

* Delete workspace button is too bright. We don't need to capture our user attention with this section.

A good example would be Vercel's UI. Button is still red, but other

![CleanShot 2025-04-28 at 15 49 10@2x](https://github.com/user-attachments/assets/286f541d-9cca-4f39-9d49-2992aec0d336)


![CleanShot 2025-04-28 at 15 39 59@2x](https://github.com/user-attachments/assets/e8f590f6-62fd-4e54-a3b3-589b7293fb99)

* When user clicks on Cancel for this dialog, the workspace still got deleted

![CleanShot 2025-04-28 at 15 56 29@2x](https://github.com/user-attachments/assets/80572ae0-a6cb-4a20-a3e0-b382ed3142d7)

danhemerlein commented:
> Overall, it looks good, thank you very much for the contribution!
> 
> But we can make it even better by addressing a few issues:
> 
> * Delete workspace button is too bright. We don't need to capture our user attention with this section.
> 
> A good example would be Vercel's UI. Button is still red, but other
> 
> ![CleanShot 2025-04-28 at 15 49 10@2x](https://private-user-images.githubusercontent.com/707733/438404741-286f541d-9cca-4f39-9d49-2992aec0d336.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDU4NzEyNDcsIm5iZiI6MTc0NTg3MDk0NywicGF0aCI6Ii83MDc3MzMvNDM4NDA0NzQxLTI4NmY1NDFkLTljY2EtNGYzOS05ZDQ5LTI5OTJhZWMwZDMzNi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNDI4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDQyOFQyMDA5MDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xOGExNDUzMzBjNzI3NzAxNjJhODkzYzQ3NzMwN2VmNTA5Y2ZlZTc5NzRmMWRmZTkwOWJiN2Q4MzYzMDM2N2Q4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.FBaIrBqNyKEVePEN2dFtRTYCUHL0qQiBtrlp-84FWWA)
> 
> ![CleanShot 2025-04-28 at 15 39 59@2x](https://private-user-images.githubusercontent.com/707733/438401690-e8f590f6-62fd-4e54-a3b3-589b7293fb99.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDU4NzEyNDcsIm5iZiI6MTc0NTg3MDk0NywicGF0aCI6Ii83MDc3MzMvNDM4NDAxNjkwLWU4ZjU5MGY2LTYyZmQtNGU1NC1hM2IzLTU4OWI3MjkzZmI5OS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNDI4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDQyOFQyMDA5MDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wNWE3MGY3YmQ0ZDVhNWFlODkzNTJhM2UwMmI0YjkwODc3YzQzNTI2OWE4MTk4YzYyNWExNmMxNmZmY2M4YjEyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.WzEoJetR-C5XGg7YtVAwIWcDNFjFm8vYUCHw9l3IDlk)
> 
> * When user clicks on Cancel for this dialog, the workspace still got deleted
> 
> ![CleanShot 2025-04-28 at 15 56 29@2x](https://private-user-images.githubusercontent.com/707733/438408114-80572ae0-a6cb-4a20-a3e0-b382ed3142d7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDU4NzEyNDcsIm5iZiI6MTc0NTg3MDk0NywicGF0aCI6Ii83MDc3MzMvNDM4NDA4MTE0LTgwNTcyYWUwLWE2Y2ItNGEyMC1hM2UwLWIzODJlZDMxNDJkNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNDI4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDQyOFQyMDA5MDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jOGUyMTYxZDUzMzYzMDcwZWVkNDNiZmE2ZTVjM2Q0Y2M3MDM4Yzc3NjY4YThjYjk1MTBlYzdlZjIxYTNkOTMwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.wyzIUzFBzZyXgq5ooX5Hku2EyVcIJ_6h5HPOLvW-D30)

Thanks for this great feedback - will address this shortly!

danhemerlein committed:
fix cancel bug deleting workspace

danhemerlein committed:
update danger zone ui

danhemerlein commented:
> Overall, it looks good, thank you very much for the contribution!
> 
> But we can make it even better by addressing a few issues:
> 
> * Delete workspace button is too bright. We don't need to capture our user attention with this section.
> 
> A good example would be Vercel's UI. Button is still red, but other
> 
> ![CleanShot 2025-04-28 at 15 49 10@2x](https://private-user-images.githubusercontent.com/707733/438404741-286f541d-9cca-4f39-9d49-2992aec0d336.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDU5NTM0NzksIm5iZiI6MTc0NTk1MzE3OSwicGF0aCI6Ii83MDc3MzMvNDM4NDA0NzQxLTI4NmY1NDFkLTljY2EtNGYzOS05ZDQ5LTI5OTJhZWMwZDMzNi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNDI5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDQyOVQxODU5MzlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xNTc0ZjRhODFiNjllYzUwYzZiMTY4YmFlMGRhOWNiZmY0ODA4OTkzZjE3OGNiNDg4OTg1ZmMyMWExM2JhOWJlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.Ca4QGuNSvrXWcdx50cT2I3i-OE7zi8QjethirQsBicw)
> 
> ![CleanShot 2025-04-28 at 15 39 59@2x](https://private-user-images.githubusercontent.com/707733/438401690-e8f590f6-62fd-4e54-a3b3-589b7293fb99.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDU5NTM0NzksIm5iZiI6MTc0NTk1MzE3OSwicGF0aCI6Ii83MDc3MzMvNDM4NDAxNjkwLWU4ZjU5MGY2LTYyZmQtNGU1NC1hM2IzLTU4OWI3MjkzZmI5OS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNDI5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDQyOVQxODU5MzlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hMTI1NGJjNzFhNGM0Y2I1ZGJmZGE1MWFmYmFlMmIxZjdlMDQzMjBkNWJhNWE0MTkyZjc2MzlmMjUyMmRkMDYyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.D2OW_B3mbPD8IPhHEVmcegOLYt5esUjRMEG_F4mrxoc)
> 
> * When user clicks on Cancel for this dialog, the workspace still got deleted
> 
> ![CleanShot 2025-04-28 at 15 56 29@2x](https://private-user-images.githubusercontent.com/707733/438408114-80572ae0-a6cb-4a20-a3e0-b382ed3142d7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDU5NTM0NzksIm5iZiI6MTc0NTk1MzE3OSwicGF0aCI6Ii83MDc3MzMvNDM4NDA4MTE0LTgwNTcyYWUwLWE2Y2ItNGEyMC1hM2UwLWIzODJlZDMxNDJkNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNDI5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDQyOVQxODU5MzlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03MjI3MWEzMmEzZGY5NjczNjhhYzdjODRmZDczOWU2Y2IwZmM5NzAwZTExYzI4MWZkYjAwODExM2RkM2JiNDMzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.MBvCG-_A0uW2qN80Q9r1BLAyBgznFi1e7wDN5d_RS7w)

addressed this feedback!

absorbb committed:
Merge branch 'newjitsu' into fork/danhemerlein/feat/delete-workspace

absorbb committed:
console: delete workspace: check that user has access to workspace
console: delete workspace: remove workspaces count and connections count requirements
console: improve settings UI

absorbb commented:
added check that user actually has access to workspace being deleted.
removed requirement that all connection must be deleted - that requirement makes it hard to restore workspace to the original state
removed requirement that user must have >1 workspaces - sometimes we get asked to delete only workspace that user has
